lib/core/filtering/simple_rules/simple_rule.rb

```diff
@@ -138,6 +138,18 @@ def to_h
         @rule_hash
       end
 
+      def try_coerce_value
+        coerced = to_float(value)
+        begin
+          coerced = to_date(value) if coerced.is_a?(String)
+        rescue ArgumentError
+          coerced = to_bool(value) if coerced.is_a?(String)
```

Comment:
I don't think we could get here if `value = "false"`

Reply:
hmm, though you have a test asserting this explicitly, so I must be wrong?

Reply:
Yes, it works for `false`.
1. `to_float` returns a string as it's not float but the implementation doesn't throw
2. `to_date` throws from the `rescue` clause because of the int conversion
3. `to_bool` is called on row 146 and works as expected `'false' => false`